### DISCUSSION TITLE
How to grab Response Headers

### DISCUSSION THREAD

**Discussion by ninjaKris** [TanStack/query#6963] [Q&A]
Is there a way to access a request's response headers via the `useQuery`?

**Comment by TkDodo** [comment]
Let me clear up a common myth: React Query doesn't do any data fetching. It only cares about Promises. So whatever value for a resolved Promise is returned from the `queryFn` will be available in the `data` field returned from `useQuery`.

So yes, it's absolutely possible to grab response headers. It's just totally _not_ tied to React Query. If you return `{ data, headers }` from the `queryFn`, you will have access to it. If not, then not.

That means it really depends on how you do the data fetching. fetch, axios, ky, graphql-request, something-else, ...

And it would work the same way as _without_ React Query. So really, I can't answer that question, because it's irrelevant from React Query's perspective.

**Reply by ninjaKris** [reply]
We're using an openapi-typescript-codegen client. I can see the headers in the Network tab, but it's undefined when I try and use {headers}

**Reply by TkDodo** [reply]
Then you need to discuss this with the lib that generates the code

**Reply by yanlimabarbosa** [reply]
hey ninjaKris, did you find a workaround around this?
